psi4/src/psi4/libmints/mcmurchiedavidson.cc

```diff
@@ -160,4 +160,77 @@ void fill_M_matrix(int maxam, int maxpow, const Point& PC, double a, double b, s
     }
 }
 
+void fill_R_matrix(int maxam, double p, const Point& P, const Point& C, std::vector<double>& R,
+                   std::shared_ptr<const libint2::FmEval_Chebyshev7<double>> fm_eval) {
+    // Generates the auxiliary integrals for Coulomb-type integrals using eq 9.9.13
+    // from Molecular Electronic-Structure Theory (10.1002/9781119019572)
+    auto PC = point_diff(P, C);
+    auto RPC = point_norm(PC);
+    double T = p * RPC * RPC;
+
+    std::vector<double> fmvals(maxam + 1);
+
+    // evaluate Boys function
+    fm_eval->eval(fmvals.data(), T, maxam);
+
+    int dim1 = maxam + 1;
+    int dim2 = dim1 * dim1 * dim1;
+    // zero out buffer
+    std::memset(&R[0], 0.0, dim1 * dim2 * sizeof(double));
```

Comment:
Can we use `std::fill` here? With standard compiler optimizations, that's as fast as `std::memset`, but it's cleaner.

Reply:
I could change it to `std::fill(R.begin(), R.begin() + dim1 * dim2, 0.0;`, I think that should work?
We don't want to zero out the entire buffer though, that was the performance bottleneck in my previous version 😄

Reply:
If it doesn't affect [performance on standard optimization](https://stackoverflow.com/a/8849789), please make the change. I initially thought that `R` was length `dim1 * dim2`. Between this and the documentation comment, it should be clearer to other non-experts that is not the case.

Reply:
Yeah I know this post on StackOverflow... I don't know if partially zeroing out would affect the performance of `std::fill`, will have to try it out.